File: CppPractice/CppLeetcode/Leetcode136.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) 
    {
        unordered_map<int,int> theMap;
        for(int num:nums)
        {
            if(theMap.find(num)==theMap.end()){theMap[num]=1;}
            else{theMap[num]+=1;}
        }
        for(auto pair:theMap)
        {if(pair.second==1){return pair.first;}}
        return -1;
    }
};

File: CSharpPractice/LeetcodePractice/Leetcode136.cs
public class Solution136 {
    public int SingleNumber(int[] nums) 
    {
        Dictionary<int,int> map = new Dictionary<int,int>();
        foreach(int num in nums)
        {
            if(!map.ContainsKey(num)){map[num]=1;}
            else{map[num]++;}
        }
        foreach(KeyValuePair<int, int> pair in map)
        {
            if(pair.Value ==1){return pair.Key;}
        }
        return -1;
    }
}